core/src/main/java/hudson/model/LargeText.java

```diff
@@ -146,27 +147,26 @@ public void close() throws IOException {
     public long writeLogTo(long start, Writer w) throws IOException {
         CountingOutputStream os = new CountingOutputStream(new WriterOutputStream(w));
 
-        Session f = source.open();
-        f.skip(start);
-
-        if(completed) {
-            // write everything till EOF
-            byte[] buf = new byte[1024];
-            int sz;
-            while((sz=f.read(buf))>=0)
-                os.write(buf,0,sz);
-        } else {
-            ByteBuf buf = new ByteBuf(null,f);
-            HeadMark head = new HeadMark(buf);
-            TailMark tail = new TailMark(buf);
-
-            while(tail.moveToNextLine(f)) {
-                head.moveTo(tail,os);
+        try (Session f = source.open()) {
+            f.skip(start);
+
+            if (completed) {
+                // write everything till EOF
+                byte[] buf = new byte[1024];
+                int sz;
+                while ((sz = f.read(buf)) >= 0)
+                    os.write(buf, 0, sz);
+            } else {
+                ByteBuf buf = new ByteBuf(null, f);
+                HeadMark head = new HeadMark(buf);
+                TailMark tail = new TailMark(buf);
+
+                while (tail.moveToNextLine(f)) {
+                    head.moveTo(tail, os);
+                }
+                head.finish(os);
             }
-            head.finish(os);
         }
-
-        f.close();
```

Comment:
Well, it makes sense to mention them and then sometimes to follow-up in further PRs. And yes, "sometimes" should happen more often